src/main/java/dk/alexandra/fresco/lib/compare/SortingProtocolBuilder.java

```diff
@@ -0,0 +1,151 @@
+package dk.alexandra.fresco.lib.compare;
+
+import java.math.BigInteger;
+
+import dk.alexandra.fresco.framework.value.OInt;
+import dk.alexandra.fresco.framework.value.SInt;
+import dk.alexandra.fresco.lib.field.integer.BasicNumericFactory;
+import dk.alexandra.fresco.lib.helper.builder.ComparisonProtocolBuilder;
+
+public class SortingProtocolBuilder extends ComparisonProtocolBuilder {
+
+	public SortingProtocolBuilder(ComparisonProtocolFactory comProvider, BasicNumericFactory bnf) {
+		super(comProvider, bnf);
+		// TODO Auto-generated constructor stub
+	}
+
+	public SInt isSorted(SInt[] values){
+		//first compare the values pairwise
+		SInt[] comparisons = new SInt[values.length-1];
+		//initialize comparisons array	
+		for (int i=0;i<comparisons.length;i++)
+			comparisons[i]=getBnf().getSInt();
+		//build parallel comparison circuit
+		beginParScope();  
+		//TODO maybe split this in chunks of reasonable sizes
+		for (int i=0;i<comparisons.length;i++) 
+			comparisons[i]=compare(values[i], values[i+1]);
+		endCurScope();
+		//then multiply the results sequentially.
+		//set result to 1
+		SInt result=getBnf().getSInt(); 
+		append(getBnf().getSInt(1,result));
+		//multiply by values from comparison
+		for (int i=0;i<comparisons.length;i++) 
+			append(getBnf().getMultCircuit(result, comparisons[i], result));
+		return result;
+	}
+
+
+	private static int FloorLog2(int value)
+	{
+		int result = -1;
+		for (int i = 1; i < value; i <<= 1, ++result);
+		return result;
+	}
+
+	final OInt minusOne = getBnf().getOInt(BigInteger.valueOf(-1L));
+	
+	public void compareAndSwap(int a,int b, SInt[] values){
```

Comment:
We already have a compare and swap implementation. I think we should use that instead, but please take a look if your code should be better than the current implementation. Find it in: dk.alexandra.fresco.lib.compare.CompareAndSwapCircuitImpl